worked_examples/geospatial/workshop_april_2024/Workshop_AprilMay2024_part2.ipynb

```diff
@@ -0,0 +1,948 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0baa4808",
+   "metadata": {},
+   "source": [
+    "To add the predictions back into a ``MapImages`` object, we simply need to load our predictions csv file as metadata.\n",
+    "\n",
+    "Since we have started a new notebook, we can create a new ``MapImages`` object by loading our patches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "85dd41cb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mapreader import load_patches"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "438d5700",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_maps = load_patches(\n",
+    "    \"./patches_100_meters/*75650661*png\", parent_paths=\"./maps/*75650661.png\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "734f324c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_maps.add_metadata(\"./infer_predictions_patch_df.csv\", ignore_mismatch=True, tree_level=\"patch\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d734c2ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_maps.add_shape()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "e0359506",
+   "metadata": {},
+   "source": [
+    "We can use the ``.show_parent()`` method to see how our predictions look on our parent map sheet (``map_74488700.png``)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "85c7f1d2",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "my_maps.show_parent(\n",
+    "    \"map_75650661.png\",\n",
+    "    column_to_plot=\"pred\",\n",
+    "    vmin=0,\n",
+    "    vmax=1,\n",
+    "    alpha=0.5,\n",
+    "    patch_border=False,\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "1ef19e32",
+   "metadata": {},
+   "source": [
+    "And the ``.convert_images()`` method to save our results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93a57626",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "parent_df, patch_df = my_maps.convert_images(save=True, save_format=\"csv\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "12eb039a",
+   "metadata": {},
+   "source": [
+    "We can also save our outputs as a ``geojson`` file using the ``.save_patches_to_geojson()`` method.\n",
+    "> _**NOTE**_: This will require you to convert your patch coordinates into a polygon format. If these aren't already available, they can be added using the ``.add_patch_polygons()`` method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c97980d7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_maps.add_patch_polygons()\n",
+    "my_maps.save_patches_to_geojson()"
```

Comment:
okay so weirdly this is because I changed the excels to csv files.

Before, we saved to excel, then loaded excels and saved to csv and so our original excel files didn't get overwritten. 
Now, we save to csv, load csv, then save to csv and then the original files are overwritten and a load of the metadata is missing (e.g. coordinates, so it can't save to geojson).